src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -742,8 +762,18 @@ ss::future<bool> sharded_store::is_compatible(
 
     auto new_valid = co_await make_valid_schema(std::move(new_schema));
 
-    auto is_compat = true;
-    for (; is_compat && ver_it != versions.end(); ++ver_it) {
+    compatibility_result result{.is_compat = true};
+
+    auto formatter = [](std::string_view rdr, std::string_view wrtr) {
+        return [rdr, wrtr](std::string_view msg) {
+            return fmt::format(
+              fmt::runtime(msg),
+              fmt::arg("reader", rdr),
+              fmt::arg("writer", wrtr));
+        };
+    };
+
+    for (; result.is_compat && ver_it != versions.end(); ++ver_it) {
```

Comment:
Good point. The reference implementation only reports the incompatibility against the first schema but includes both forward and backward incompatibilities. That's also what we do here and I think that makes sense because it can become confusing quickly if multiple incompatibilities are reported against multiple schemas.

One difference though is that the reference implementation walks the schemas in the reverse direction (showing incompatibilities against the latest schema first). That might be useful to pick up as a user might have an easier time comparing recent schemas than old ones. Wdyt?

Reply:
>showing incompatibilities against the latest schema first...might be useful to pick up

Agree. Mirroring the reference impl is the name of the game IMO. I don't recall doing much testing with respect to checking multiple versions, so any convergence or divergence is basically accidental.

Reply:
Fixed